src/hardware/mixer.cpp

```diff
@@ -784,17 +838,17 @@ void MixerChannel::AddStretched(uint16_t len, int16_t *data)
 		const auto new_pos = index >> FREQ_SHIFT;
 		if (pos != new_pos) {
 			pos = new_pos;
-			//Forward the previous sample
-			prev_sample[0] = data[0];
+			// Forward the previous sample
+			prev_frame.left = data[0];
 			data++;
 		}
-		const auto diff = data[0] - prev_sample[0];
+		const auto diff = data[0] - static_cast<int16_t>(prev_frame.left);
```

Comment:
Are we sure the cast won't wrap the left sample, if it's beyond 16bit (or into the larger int32/float range?)

Reply:
Can always use `check_cast<smaller_type>(val)`, which add range-assert to the value prior to casting. Zero performance hit for release builds.

Reply:
Taking a step back, `AddStretched` is only used in a single place in `sblaster.cpp`. I wonder what's so special about the Sound Blaster emulation that warrants the use of this method. Probably nothing, therefore I'm inclined to remove it completely, use the regular `AddSamples`, and see how that goes.

Reply:
(But the SB emulations uses it in conjunction with this buffer: `int16_t data[DSP_DACSIZE + 1] = {};`, so currently it's fine).

Reply:
Just was thinking through the pathway - if a channel adds large floats using `AddSamples_sfloat()`, then this sets up the `prev_frame.left` value to be out-of-range when the channel next calls `AddStretched()`.

This PR mixes adding floats + int16s from the same channel (and rightly so!), so it probably makes sense just to assert that `prev_frame.left` is in-range before the cast.

Reply:
I'll try to remove it first altogether 😄 I don't like it 😄